src/cloud_manager.ts

```diff
@@ -30,17 +33,24 @@ export interface CloudInstance {
 export default class CloudManager {
     private instanceTracker: InstanceTracker;
     private oracleInstanceManager: OracleInstanceManager;
+    private customInstanceManager: CustomInstanceManager;
     private shutdownManager: ShutdownManager;
     private audit: Audit;
     private isDryRun: boolean;
 
     constructor(options: CloudManagerOptions) {
         this.isDryRun = options.isDryRun;
-        this.oracleInstanceManager = new OracleCloudManager({
-            isDryRun: options.isDryRun,
-            ociConfigurationFilePath: options.ociConfigurationFilePath,
-            ociConfigurationProfile: options.ociConfigurationProfile,
-        });
+        if (options.cloudProvider === 'oracle') {
```

Comment:
I believe eventually we may allow for multiple cloud providers here, and let the group cloud setting control which manager to use.  For now this is fine, but for future we may consider instantiating all possible managers for which we have configuration options specified.